File: abc/232/c.php
<?php
function strings() { return explode(' ', trim(fgets(STDIN))); }
function ints() { return array_map('intval', strings()); }
function doubles() { return array_map('doubleval', strings()); }
function chmax(&$a, $b) { if($a < $b) { $a = $b; return true; } return false; }
function chmin(&$a, $b) { if($a > $b) { $a = $b; return true; } return false; }
function gcd($a, $b) { return ($a % $b) ? gcd($b, $a % $b) : $b; }
function lcm($a, $b) { return intdiv($a, gcd($a, $b)) * $b; }
function swap(&$a, &$b) { list($a, $b) = [$b, $a]; }
function sub_sort(&$a, $i, $j) { $b = array_slice($a, $i, $j - $i); sort($b); foreach($b as $k => $v){ $a[$k + $i] = $v; }}
function sub_reverse(&$a, $i, $j) { $b = array_slice($a, $i, $j - $i); foreach(array_reverse($b) as $k => $v){ $a[$k + $i] = $v; }}
function next_permutation(&$a) { $n = count($a); for($i = $n - 2; $i >= 0; --$i) { if($a[$i] < $a[$i + 1]){ for($j = $n - 1;; --$j){ if($a[$i] < $a[$j]){ swap($a[$i], $a[$j]); sub_reverse($a, $i + 1, $n); return true; }}}} return false; }
function __builtin_popcount($i) { if($i){ return ($i & 1) + __builtin_popcount($i >> 1); } return 0; }
function mod_power($a, $n, $m) { if($n === 0) return 1; if($n === 1) return $a % $m; if($n % 2 === 1) return ($a * mod_power($a, $n - 1, $m)) % $m; return pow(mod_power($a, $n >> 1, $m), 2) % $m;}
function mod_inverse($a, $m) { $b = $m; $u = 1; $v = 0; while($b){ $t = intdiv($a, $b); $a -= $t * $b; swap($a, $b); $u -= $t * $v; swap($u, $v); } $u += $m; return $u % $m; }
function mod_combination($a, $b, $m) { $t = 1; for($i = 0; $i < $b; ++$i){ $t *= $a - $i; $t %= $m; $t *= mod_inverse($i + 1, $m); $t %= $m; } return $t; }
function mod_multi_combination($a, $b, $m) { if($a === 0 && $b === 0) return 1; if($a <= 0 || $b < 0) return 0; return mod_combination($a + $b - 1, $b, $m); }
function run_length_encode($s) { $n = strlen($s); $t = ''; $a = []; $j = -1; for($i = 0; $i < $n; ++$i){ if($t !== $s[$i]) { ++$j; $t = $s[$i]; $a[$j] = [$t, 1]; } else { ++$a[$j][1]; }} return $a; }
function run_length_decode($a) { $s = ''; foreach($a as $v){ for($i = 0; $i < $v[1]; ++$i){ $s .= $v[0]; }} return $s; }
const _DIF = [[-1, 0], [0, 1], [1, 0], [0, -1]];
const _INF = 1001001001;
const _LLINF = 1001001001001001001;
const _MOD = 1000000007;

function main()
{
    list($n, $m) = ints();

    $takahashi = array_fill(0, $n, array_fill(0, $n, false)); // ボールの連結情報を格納した行列
    $aoki = array_fill(0, $n, array_fill(0, $n, false)); // ボールの連結情報を格納した行列

    for($i = 0; $i < $m; ++$i)
    {
        list($a, $b) = ints();

        --$a; --$b; // 0-indexedにする

        $takahashi[$a][$b] = true;
        $takahashi[$b][$a] = true;
    }

    for($i = 0; $i < $m; ++$i)
    {
        list($c, $d) = ints();

        --$c; --$d; // 0-indexedにする

        $aoki[$c][$d] = true;
        $aoki[$d][$c] = true;
    }

    $p = range(0, $n - 1);
    $is_ok = false;

    do
    {
        $is_same = true;

        for($i = 0; $i < $n; ++$i)
        {
            for($j = 0; $j < $n; ++$j)
            {
                if($takahashi[$i][$j] !== $aoki[$p[$i]][$p[$j]])
                {
                    $is_same = false;
                }
            }
        }

        if($is_same)
        {
            $is_ok = true;
            break;
        }
    }
    while(next_permutation($p));

    printf("%s\n", $is_ok ? 'Yes' : 'No');
}

main();

File: abc/232/c.cpp
#include <bits/stdc++.h>
using namespace std;

#define _overload3(_1,_2,_3,name,...) name
#define _REP(i,n) REPI(i,0,n)
#define REPI(i,a,b) for(int i=int(a);i<int(b);++i)
#define REP(...) _overload3(__VA_ARGS__,REPI,_REP,)(__VA_ARGS__)
#define ALL(a) (a).begin(),(a).end()
#define ALLR(a) (a).rbegin(),(a).rend()
typedef long long ll;
typedef long double ld;
const int MOD = 1e9+7;
const int INF = 2e9;
const long long LLINF = 2e18;
template<class T> bool chmax(T &a, const T &b) { if (a<b) { a=b; return 1; } return 0; }
template<class T> bool chmin(T &a, const T &b) { if (a>b) { a=b; return 1; } return 0; }
ll gcd(ll x, ll y){ return (x % y) ? gcd(y, x % y) : y; }
ll lcm(ll x, ll y){ return x / gcd(x, y) * y; }

using Graph = vector<vector<int>>;

int main(){
    int n, m;
    cin >> n >> m;
    // vector<int> a(m), b(m), c(m), d(m);
    Graph ta(n), ao(n);
    REP(i, m){
        int a, b;
        // cin >> a[i] >> b[i];
        cin >> a >> b;
        // --a[i], --b[i];
        --a, --b;
        ta[a].push_back(b);
        ta[b].push_back(a);
    }
    REP(i, m){
        // cin >> c[i] >> d[i];
        // --c[i], --d[i];
        int c, d;
        cin >> c >> d;
        --c, --d;
        ao[c].push_back(d);
        ao[d].push_back(c);
    }
    vector<int> p(n);
    REP(i, n){
        p[i] = i;
    }
    bool is_ok = true;
    do{
        // bool test = false;
        // if(p[0] == 2 && p[1] == 1 && p[2] == 0 && p[3] == 3){
        //     test = true;
        //     cout << "test" << endl;
        // }
        is_ok = true;
        REP(i, n){
            // if(test){
            //     cout << ta[i].size() << ' ' << ao[p[i]].size() << endl;
            // }
            if(ta[i].size() != ao[p[i]].size()){
                is_ok = false;
                break;
            }
            set<int> s;
            for(int v : ta[i]){
                s.insert(v);
                // if(test){
                //     cout << v << endl;
                // }
            }
            for(int v : ao[p[i]]){
                s.insert(p[v]);
                // if(test){
                //     cout << v << endl;
                // }
            }
            // if(test){
            //     cout << ta[i].size() << ' ' << s.size() << endl;
            // }
            if(ta[i].size() != s.size()){
                is_ok = false;
                break;
            }
        }
        if(is_ok){
            break;
        }
    }while(next_permutation(ALL(p)));
    if(is_ok){
        cout << "Yes" << endl;
    }
    else{
        cout << "No" << endl;
    }
}